pandas/core/arrays/datetimelike.py

```diff
@@ -1830,16 +1830,21 @@ def validate_endpoints(closed):
     left_closed = False
     right_closed = False
 
-    if closed is None:
+    if closed is None or closed is True or closed == "neither":
```

Comment:
I would use the standard value first, i.e. `if closed == "neither" or (# additional conditionals for backwards compat)`.

Also I think you mean `closed=="both"` here and not `closed="neither"`.